dot/sync/disjoint_block_set_test.go

```diff
@@ -0,0 +1,313 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_disjointBlockSet_addBlock(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name             string
+		disjointBlockSet disjointBlockSet
+		block            *types.Block
+		err              error
+	}{
+		{
+			name: "add block beyond capacity",
+			block: &types.Block{
+				Header: types.Header{
+					Number: 1,
+				},
+			},
+			err: errors.New("cannot add block; set is at capacity"),
+		},
+		{
+			name: "add block",
+			disjointBlockSet: disjointBlockSet{
+				limit:            1,
+				blocks:           make(map[common.Hash]*pendingBlock),
+				timeNow:          time.Now,
+				parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
+			},
+			block: &types.Block{
+				Header: types.Header{
+					Number: 1,
+				},
+			},
+			err: nil,
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := tt.disjointBlockSet.addBlock(tt.block)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_disjointBlockSet_addHeader(t *testing.T) {
+	t.Parallel()
+
+	mockTime := func() time.Time { return time.Unix(1001, 0) }
```

Comment:
nit Define this inline in the `add header` test case since it's only used there

Reply:
Updated.